apps/voting/app/src/vote-description-utils.js

```diff
@@ -0,0 +1,115 @@
+import React, { useMemo } from 'react'
+import styled from 'styled-components'
+import { GU, useTheme } from '@aragon/ui'
+import LocalIdentityBadge from './components/LocalIdentityBadge/LocalIdentityBadge'
+import LocalLabelAppBadge from './components/LocalIdentityBadge/LocalLabelAppBadge'
+
+export function renderDescription(path) {
+  const theme = useTheme()
+  const description = useMemo(() => {
+    return path
+      ? path.map((step, index) =>
+          renderForwardingStep(step, [index + 1], theme)
+        )
+      : ''
+  }, [path])
+  return <React.Fragment>{description}</React.Fragment>
+}
+
+function renderForwardingStep(step, depth, theme) {
+  const app =
+    step.name || step.identifier ? (
+      <React.Fragment key={0}>
+        {step.name && step.name}
+        {step.identifier ? ` (${step.identifier})` : ''}
+        <span
+          css={`
+            padding-right: ${1 * GU}px;
+          `}
+        >
+          :
+        </span>
+      </React.Fragment>
+    ) : (
+      <React.Fragment key={0}>
+        <LocalIdentityBadge
+          compact
+          entity={step.to}
+          css={`
+            padding-right: 0;
+          `}
+        />
+        <span
+          css={`
+            padding-right: ${1 * GU}px;
+          `}
+        >
+          :
+        </span>
+      </React.Fragment>
+    )
+
+  let description = ''
+  if (step.annotatedDescription) {
+    description = step.annotatedDescription.map((descriptionLine, index) => {
+      if (descriptionLine.type == 'address') {
+        return (
+          <LocalIdentityBadge
+            key={index + 1}
+            compact
+            entity={descriptionLine.value}
+          />
+        )
+      }
+      return (
+        <span key={index + 1}>
+          {descriptionLine.value.description || descriptionLine.value}
```

Comment:
We should check that `descriptionLine.value` is a string first, since it can also be an object (app or permission), making it crash (e.g. with http://localhost:3000/?#/dcl/0xbc1863a593aaebb40b4f43665de30174c9d3fe29/vote/17/).